### ISSUE TITLE
JSON transport fails silently

### ISSUE THREAD

**Issue by avaly** [JaniAnttonen/winston-loki#4]
It seems the JSON request is failing silently.

```json
reqBody={"streams":[{"labels":"{job=\"undefined\", level=\"info\"}","entries":[{"line":"Listening on port: 8000 "}]}]}
```

The `got` request is rejected with this error:

```
{ HTTPError: Response code 500 (Internal Server Error)
    at EventEmitter.emitter.on (/home/dev/node_modules/got/source/as-promise.js:74:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
  name: 'HTTPError',
  host: 'localhost:3100',
  hostname: 'localhost',
  method: 'POST',
  path: '/api/prom/push',
  socketPath: undefined,
  protocol: 'http:',
  url: 'http://localhost:3100/api/prom/push',
  gotOptions: 
   { path: '/api/prom/push',
     protocol: 'http:',
     slashes: true,
     auth: null,
     host: 'localhost:3100',
     port: '3100',
     hostname: 'localhost',
     hash: null,
     search: null,
     query: null,
     pathname: '/api/prom/push',
     href: 'http://localhost:3100/api/prom/push',
     retry: 
      { retries: [Function],
        methods: [Object],
        statusCodes: [Object],
        errorCodes: [Object] },
     headers: 
      { 'user-agent': 'got/9.6.0 (https://github.com/sindresorhus/got)',
        'content-type': 'application/json',
        'accept-encoding': 'gzip, deflate',
        'content-length': 110 },
     hooks: 
      { beforeRequest: [],
        beforeRedirect: [],
        beforeRetry: [],
        afterResponse: [],
        beforeError: [],
        init: [] },
     decompress: true,
     throwHttpErrors: true,
     followRedirect: true,
     stream: false,
     form: false,
     json: false,
     cache: false,
     useElectronNet: false,
     body: '{"streams":[{"labels":"{job=\\"undefined\\", level=\\"info\\"}","entries":[{"line":"Listening on port: 8000 "}]}]}',
     method: 'POST' },
  statusCode: 500,
  statusMessage: 'Internal Server Error',
  headers: 
   { 'content-type': 'text/plain; charset=utf-8',
     'x-content-type-options': 'nosniff',
     date: 'Tue, 29 Jan 2019 08:47:47 GMT',
     'content-length': '52',
     connection: 'close' },
  body: 'rpc error: code = Unknown desc = Entry out of order\n' }
```

The same log entries with the protobuf gets stored correctly in Loki.

This may be an issue on Loki's API side. Feel free to close if this is true.

**Comment by JaniAnttonen** [comment]
Thank you for the issue!

This is due to https://github.com/grafana/loki/issues/168
I'm working on an update that should fix this: #5

**Comment by JaniAnttonen** [comment]
It seems that there has been a PR in Loki that should've fixed these problems. Will do some testing and then close this if that's the case: https://github.com/grafana/loki/pull/288
